source/contribute/maintainer.rst

```diff
@@ -59,16 +67,167 @@ Optionally, add any additional contributor/maintainer forks, e.g.:
+        fetch = +refs/heads/*:refs/remotes/doctrine/*
+        # add this:
+        fetch = +refs/pull/*/head:refs/remotes/doctrine/pr/*
+
+Distinguishing features, bugs and critical fixes
+------------------------------------------------
+
+The primary role of a maintainer, besides being also a contributor,
+is to sort incoming proposals by their category:
+
+-  **new features** are additions that provide new API or new behavior
+   that was previously not exposed by the project
+-  **improvements** are additions that improve existing API by making
+   it more clear, by improving the usability and performance, or by
+   verifying existing behavior via refactoring, new tests or static
+   analysis.
+-  **bug fixes** are changes to the codebase that do correct invalid
+   behaviour.
+-  **critical fixes** are changes to the codebase that correct invalid
+   behavior that was erroneously introduced, and prevents installation
+   or usage of the library by a very large portion of the community.
+-  **security fixes** are changes to the codebase that correct existing
+   behavior in the codebase that may lead to substantial financial or
+   personal damage to consumers of the packages due to malicious
+   attack vectors. Read about our `Security Policy <https://www.doctrine-project.org/policies/security.html>`_
+-  **deprecations** are changes to the codebase that do mark existing
+   API as "to be removed in future". Read about our `Deprecation Policy <https://www.doctrine-project.org/policies/deprecation.html>`_
+-  **backwards-compatibility breakages** (or **BC breaks** in short)
+   are modifications to the existing API or implementation that would
+   result in downstream users having to correct their software to
+   adapt to the new changes. Maintainers should also prevent any
+   unnecessary BC breaks, and always evaluating if it is worth
+   introducing them.
+
+Stability and Semantic Versioning
+---------------------------------
+
+A maintainer must also always consider that any proposal, regardless
+how well tested and verified it could be, brings in some instability.
+In order to reduce the amount of defects and/or regressions reaching
+downstream users, a maintainer must therefore always consider
+carefully where a patch may land.
+
+Packages controlled by the Doctrine organisation are to follow
+`Semantic Versioning (SemVer) <https://semver.org/spec/v2.0.0.html>`_
+rules, with the famous ``MAJOR.MINOR.PATCH`` naming scheme.
+
+This effectively means:
+
+- **PATCH** only contains **bug fixes**, **security fixes** and
+  **critical fixes**
+- **MINOR** can contain everything that is in **PATCH** plus
+  **new features**, **improvements** and **deprecations**
+- **MAJOR** can contain all of the above plus **BC breaks**
+
+Whilst it is possible to automate some of these decisions, humans
+are still better at categorising these changes due to the amount of
+nuances that are involved in the software development process.
+
 Branching Model
```

Comment:
This section needs an update after the decision at the EUFOSSA hackathon. See https://github.com/doctrine/automatic-releases#todos for the process.